CefSharp.WinForms/Internals/MovingListener.cs

```diff
@@ -0,0 +1,155 @@
+            parent_ParentChanged(ParentControl, null);
+        }
+
+        private void parent_ParentChanged(object sender, EventArgs e)
+        {
+            Control control = (Control)sender;
+            Form oldForm = ParentForm;
+            Form newForm = control.FindForm();
+            if (oldForm == null
+                || newForm == null
+                || oldForm.Handle != newForm.Handle
+                )
+            {
+                if (this.Handle != IntPtr.Zero)
+                {
+                    ReleaseHandle();
+                }
+                if (oldForm != null)
+                {
+                    oldForm.HandleCreated -= OnHandleCreated;
+                    oldForm.HandleDestroyed -= OnHandleDestroyed;
+                }
+                ParentForm = newForm;
+                if (newForm != null)
+                {
+                    OnHandleCreated(newForm, null);
+                }
+            }
+        }
+
+        private void OnHandleCreated(object sender, EventArgs e)
+        {
+            AssignHandle(((Form)sender).Handle);
+            Kernel32.OutputDebugString(String.Format("Attaching '{1:x}' to form '{0:x}'\r\n", ParentForm.Handle, ParentControl.Handle));
+        }
+
+        private void OnHandleDestroyed(object sender, EventArgs e)
+        {
+            ReleaseHandle();
+        }
+        protected override void WndProc(ref Message m)
+        {
+            bool care = false;
+
+            // Negative initial values keeps the compiler quiet:
+            int x = -1;
+            int y = -1;
+
+            // Listen for operating system messages 
+            switch (m.Msg)
+            {
+            // WM_MOVING:
+            case 0x0216:
+                movingRectangle = (Rectangle)Marshal.PtrToStructure(m.LParam, typeof(Rectangle));
+                x = movingRectangle.Left;
+                y = movingRectangle.Top;
+                care = true;
+                break;
+            // WM_MOVE
+            case 0x3:
+                x = (m.LParam.ToInt32() & 0xffff);
+                y = ((m.LParam.ToInt32() >> 16) & 0xffff);
+                care = true;
+                break;
+            }
+
+            // Only notify about movement if:
+            // * ParentForm Created
+            // * ParentControl Created
+            // * ParentForm Actually moved
+            // * Not currently moving (on the UI thread only of course)
+            // * The current WindowState is Normal.
+            //      This check is to simplify the effort here.
+            //      Other logic already handles the maximize/minimize
+            //      cases just fine.
+            // You might consider checking ParentControl.Visible and
+            // not notifying our parent control if the parent control isn't visible.
+            // However, if you do that, the non-Visible CEF tab will still
+            // have any SELECT drop downs rendering where they shouldn't.
+            if (care
```

Comment:
Hrm. ParentForm/ParentControl created might not be needed any longer. I'll give that a try.

The window state/actual movement check probably should stay.

The note about NOT checking .Visible on purpose probably needs to stay.